src/app/DataAccessLayer/dalMain.cs

```diff
@@ -11,21 +12,28 @@ public partial class DAL : IDAL
     {
         private CosmosDetails _cosmosDetails = null;
 
+
         /// <summary>
         /// Data Access Layer Constructor
         /// </summary>
-        /// <param name="cosmosUrl">CosmosDB URL</param>
+        /// <param name="cosmosUrl">CosmosDB Url</param>
         /// <param name="cosmosKey">CosmosDB connection key</param>
         /// <param name="cosmosDatabase">CosmosDB Database</param>
         /// <param name="cosmosCollection">CosmosDB Collection</param>
-        public DAL(string cosmosUrl, string cosmosKey, string cosmosDatabase, string cosmosCollection)
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Reliability", "CA2000:Dispose objects before losing scope", Justification = "objects are used by DI")]
+        public DAL(Uri cosmosUrl, string cosmosKey, string cosmosDatabase, string cosmosCollection)
         {
+            if (cosmosUrl == null)
+            {
+                throw new ArgumentNullException(nameof(cosmosUrl));
+            }
+
             _cosmosDetails = new CosmosDetails
             {
                 CosmosCollection = cosmosCollection,
                 CosmosDatabase = cosmosDatabase,
                 CosmosKey = cosmosKey,
-                CosmosUrl = cosmosUrl
+                CosmosUrl = cosmosUrl.AbsoluteUri
             };
 
             // create the CosmosDB client and container
```

Comment:
In Visual Studio, I still the following message left calling out line 40.  Not sure if you see it too or if we need to suppress/address it. 
Message	IDE0067	Disposable object created by 'OpenAndTestCosmosClient(cosmosUrl, cosmosKey, cosmosDatabase, cosmosCollection)' is never disposed.

More info:
Use recommended dispose pattern to ensure that locally scoped disposable objects are disposed on all paths. If possible, wrap the creation within a 'using' statement or a 'using' declaration. Otherwise, use a try-finally pattern, with a dedicated local variable declared before the try region and an unconditional Dispose invocation on non-null value in the 'finally' region, say 'x?.Dispose()'. If the object is explicitly disposed within the try region or the dispose ownership is transfered to another object or method, assign 'null' to the local variable just after such an operation to prevent double dispose in 'finally'

Reply:
I don't get that. CA2000 is suppressed in line 23 because we can't dispose of the object because it's used by DI. Not sure why you're getting a message on that one.  It doesn't show up in docker build either.

need to figure out why fxcop is showing different results ...

Reply:
It went away after updating Visual Studio...